File: Python/exercicio9.py
from abc import ABC, abstractmethod

class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def imprimir(self):
        print("Imprimindo relatório")

class Contrato(Imprimivel):
    def imprimir(self):
        print("Imprimindo contrato")

itens = [Relatorio(), Contrato()]
for item in itens:
    item.imprimir()

File: Java/exercicio9.java
interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    public void imprimir() {
        System.out.println("Imprimindo relatório");
    }
}

class Contrato implements Imprimivel {
    public void imprimir() {
        System.out.println("Imprimindo contrato");
    }
}

public class Main {
    public static void main(String[] args) {
        Imprimivel[] itens = {new Relatorio(), new Contrato()};
        for (Imprimivel item : itens) {
            item.imprimir();
        }
    }
}
